core/src/main/java/org/elasticsearch/rest/action/cat/RestNodeAttrsAction.java

```diff
@@ -0,0 +1,131 @@
+
+public class RestNodeAttrsAction extends AbstractCatAction {
+
+    @Inject
+    public RestNodeAttrsAction(Settings settings, RestController controller, Client client) {
+        super(settings, controller, client);
+        controller.registerHandler(GET, "/_cat/nodeattrs", this);
+    }
+
+    @Override
+    void documentation(StringBuilder sb) {
+        sb.append("/_cat/nodeattrs\n");
+    }
+
+    @Override
+    public void doRequest(final RestRequest request, final RestChannel channel, final Client client) {
+        final ClusterStateRequest clusterStateRequest = new ClusterStateRequest();
+        clusterStateRequest.clear().nodes(true);
+        clusterStateRequest.local(request.paramAsBoolean("local", clusterStateRequest.local()));
+        clusterStateRequest.masterNodeTimeout(request.paramAsTime("master_timeout", clusterStateRequest.masterNodeTimeout()));
+
+        client.admin().cluster().state(clusterStateRequest, new RestActionListener<ClusterStateResponse>(channel) {
+            @Override
+            public void processResponse(final ClusterStateResponse clusterStateResponse) {
+                NodesInfoRequest nodesInfoRequest = new NodesInfoRequest();
+                nodesInfoRequest.clear().jvm(false).os(false).process(true);
+                client.admin().cluster().nodesInfo(nodesInfoRequest, new RestActionListener<NodesInfoResponse>(channel) {
+                    @Override
+                    public void processResponse(final NodesInfoResponse nodesInfoResponse) {
+                        NodesStatsRequest nodesStatsRequest = new NodesStatsRequest();
+                        nodesStatsRequest.clear().jvm(false).os(false).fs(false).indices(false).process(false);
+                        client.admin().cluster().nodesStats(nodesStatsRequest, new RestResponseListener<NodesStatsResponse>(channel) {
+                            @Override
+                            public RestResponse buildResponse(NodesStatsResponse nodesStatsResponse) throws Exception {
+                                return RestTable.buildResponse(buildTable(request, clusterStateResponse, nodesInfoResponse, nodesStatsResponse), channel);
+                            }
+                        });
+                    }
+                });
+            }
+        });
+    }
+
+    @Override
+    Table getTableWithHeader(final RestRequest request) {
+        Table table = new Table();
+        table.startHeaders();
+        table.addCell("node", "default:true;alias:name;desc:node name");
+        table.addCell("id",   "default:false;alias:id,nodeId;desc:unique node id");
+        table.addCell("pid",  "default:false;alias:p;desc:process id");
+        table.addCell("host", "alias:h;desc:host name");
+        table.addCell("ip",   "alias:i;desc:ip address");
+        table.addCell("port", "default:false;alias:po;desc:bound transport port");
+        table.addCell("attr.name");
+        table.addCell("attr.value");
```

Comment:
Now that I'm seeing these, I wonder if the default names should be `attr` and `value`. We could add aliases if we need longer. Since it's such a small API it's probably fine with the short versions.